src/Data/Bits/Extras.hs

```diff
@@ -38,22 +43,57 @@ import Data.Word
 import Foreign.Ptr
 import Foreign.Storable
 import GHC.Base
+#if __GLASGOW_HASKELL__ >= 900
+import Control.Exception
+import GHC.Integer.Logarithms
+#endif
 
 -- $setup
 -- >>> import Data.Bits (Bits(..))
 -- >>> import Data.Word (Word)
 
--- TODO: generalize to 64 bits, etc.
+-- | Calculate the integer base 2 logarithm. The argument must be strictly
+-- positive. Since ghc-9.0, the augment type is generalized from 'Word32' to
+-- any 'Integral' type.
+#if __GLASGOW_HASKELL__ >= 900
+
+log2 :: Integral a => a -> Int
```

Comment:
Move these Haddocks to be next to the definition of `log2`. Otherwise, they'll be attached to the wrong function on old versions of GHC:

```suggestion
#if __GLASGOW_HASKELL__ >= 900

-- | Calculate the integer base 2 logarithm. The argument must be strictly
-- positive. On GHC 9.0 or later, the argument type is generalized from 'Word32' to
-- any 'Integral' type.
log2 :: Integral a => a -> Int
```

I also fixed some typos in the Haddocks above (e.g., `s/augment/argument/`).